source/WindowScroller/WindowScroller.js

```diff
@@ -145,10 +162,18 @@ export default class WindowScroller extends React.PureComponent<Props, State> {
   }
 
   componentWillUnmount() {
-    if (this.props.scrollElement) {
-      unregisterScrollListener(this, this.props.scrollElement);
+    const scrollElement = this.props.scrollElement;
+    if (scrollElement) {
+      unregisterScrollListener(this, scrollElement);
+    }
+    if (scrollElement === window) {
+      window.removeEventListener('resize', this._onResize, false);
+    } else if (this._detectElementResize && scrollElement) {
```

Comment:
This doesn't handle the case where the `scrollElement` prop changes from `window` to `!window` (or vice versa) within the life of the instance. I don't know if that's ever going to actually happen though. 😄

Reply:
Done